src/util/rocksdb/properties.rs

```diff
@@ -466,6 +479,255 @@ impl DecodeProperties for UserCollectedProperties {
     }
 }
 
+pub struct IndexHandlesIterator<'a> {
+    iter: Iter<'a, Vec<u8>, Vec<u64>>,
+    idx: usize,
+}
+
+impl<'a> IndexHandlesIterator<'a> {
+    fn new(iter: Iter<'a, Vec<u8>, Vec<u64>>, idx: usize) -> IndexHandlesIterator<'a> {
+        IndexHandlesIterator { iter, idx }
+    }
+}
+
+impl<'a> Iterator for IndexHandlesIterator<'a> {
+    // type Item = (&'a [u8], &'a IndexHandle);
+    type Item = (&'a [u8], u64);
+
+    fn next(&mut self) -> Option<(&'a [u8], u64)> {
+        match self.iter.next() {
+            None => None,
+            Some((k, handles)) => Some((k, handles[self.idx])),
+        }
+    }
+}
+
+/// `OffsetsHandle` manages the multiple properties's offsets of the block in the file.
+#[derive(Debug, Default)]
+pub struct OffsetsHandle {
+    // key=>(offsets list)
+    pub offsets: BTreeMap<Vec<u8>, Vec<u64>>,
+    // the number of properties stored.
+    size: usize,
```

Comment:
I think this `size` is not easy to understand and can be error-prone, can you come up with a better solution?